ac-62 implement product transactions api

ExtraGravity opened the pull request:
Create database calls to get product transactions per type.
Different database types are hard coded, couldn't figure out a general
function for any kind of product type, we should merge for now for
time's sake.
Edited DB init script a bit to fix some typos.

ExtraGravity committed:
ac-62 implement product transactions api

Create database calls to get product transactions per type.
Attributes are hardcoded, but not types. So if we add new product types
without adding new attributes, no code changes are needed.
Edited DB init script a bit to fix some typos.

beaesguerra commented on src/database/ProductTransactions/index.ts:
```diff
@@ -0,0 +1,98 @@
+  lastmodified: string;
+}
+
+interface ProductTransactionAttribute {
```
need db after this?

beaesguerra commented on src/database/ProductTransactions/index.ts:
```diff
@@ -0,0 +1,98 @@
+}
+
+interface ProductTransaction {
+  uuid: string;
```
i just realized we aren't specifying what type the product is. should we add that?

beaesguerra commented on src/routers/transactions/products.ts:
```diff
@@ -51,8 +52,9 @@ const router = createRouter();
-}, authorized(UserType.ADMIN));
+  const result = await ProductTransactionsDb.getProductTransactions(req.params.type);
+  res.status(StatusCode.OK).send(result);
+});
```
was this supposed to have no permissions?

ExtraGravity committed:
Merge branch 'master' into enoch/ac-62-get-product-transactions

ExtraGravity committed:
ac-62 respond to bea's review #63

Return product type as well.
Fix linter errors.
Add Db suffix to db interface types.
